### PR TITLE
Improve gradle scripts

### PR DESCRIPTION
**Pull request by naftalmm**
This PR mainly focused on complete migration to [Task Configuration Avoidance API](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html) in gradle buildscripts, which is a recommended way to work with tasks in modern Gradle. It is available since Gradle 4.9, while Kotlin Gradle plugin 1.4.10 requires Gradle 5.4+, so it could be used in examples without any additional notes (actually it was already used somethere).

Also moved to usage of some overloaded methods variants in Gradle API, making kotlin code more idiomatic. 
And fix a couple of groovy scripts to make them fully equivalent with their kotlin counterparts.

### CODE REVIEW

File: pages/docs/reference/inline-classes.md [hunk 1]
```diff
@@ -199,7 +199,7 @@ To remove the warning you have to opt in to the usage of this feature by passing
 <div class="sample" markdown="1" theme="idea" mode="groovy" data-lang="groovy">
 
 ```groovy
-compileKotlin {
+tasks.withType(KotlinCompile).configureEach {
```
Comment (h0tk3y) [inline pages/docs/reference/inline-classes.md:202]: Maybe we should replace this and other occurrences where a language feature is enabled in a task's compiler args with `kotlin.sourceSets.all { languageSettings.enableLanguageFeature("InlineClasses") }`.
@p7nov WDYT?

File: pages/docs/reference/using-gradle.md [hunk 2]
```diff
@@ -610,8 +610,10 @@ It is also possible to configure all Kotlin compilation tasks in the project:
 <div class="sample" markdown="1" mode="groovy" theme="idea" data-lang="groovy">
 
 ```groovy
-tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
-    kotlinOptions { //... }
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
```
Comment (h0tk3y) [inline pages/docs/reference/using-gradle.md:613]: Note that the snippet that uses the FQN inline is a bit easier to copy-paste, so I'm not sure we should switch to using imports.

**Review by h0tk3y — APPROVED** [review]

File: pages/docs/reference/inline-classes.md (hunk 1 above)
Comment (p7nov) [reply pages/docs/reference/inline-classes.md:202]: It's a good idea, let's do.

**Comment by p7nov** [comment]
Thank you for this update!

**Pull request merged**